src/__tests__/TreeViewMetadata.test.tsx

```diff
@@ -46,10 +50,10 @@ const initialTreeNode = {
   ],
 };
 
-const mapDataType = flattenTree(initialTreeNode);
+const mapDataType = flattenTree<Metadata>(initialTreeNode);
```

Comment:
I’d like to make sure that the flatten tree function can automatically infer the meta-data type. So please see what happens if you delete the type parameter.